build: move e2e dependencies into `e2e/go.mod`

nixpanic opened the pull request:
Several packages are only used while running the e2e suite. These
packages are less important to update, as the they can not influence the
final executable that is part of the Ceph-CSI container-image.

By moving these dependencies out of the main Ceph-CSI go.mod, it is
easier to identify if a reported CVE affects Ceph-CSI, or only the
testing (like most of the Kubernetes CVEs).

nixpanic committed:
util: move kernel version functions to pkg/util/kernel

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
util: move EncryptionType(s) to pkg/util/crypto

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
build: move e2e dependencies into e2e/go.mod

Several packages are only used while running the e2e suite. These
packages are less important to update, as the they can not influence the
final executable that is part of the Ceph-CSI container-image.

By moving these dependencies out of the main Ceph-CSI go.mod, it is
easier to identify if a reported CVE affects Ceph-CSI, or only the
testing (like most of the Kubernetes CVEs).

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
ci: add e2e module to Dependabot configuration

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
build: no need to list all go modules

It is not clear why listing all modules is useful. Some modules have
internal versions (Kubernetes modules), and the command reports those as
a warning with an error exit code.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Rakshith-R commented on e2e/rbd.go:
```diff
@@ -23,7 +23,8 @@ import (
 	"time"
 
-	"github.com/ceph/ceph-csi/internal/util"
+	"github.com/ceph/ceph-csi/pkg/util/crypto"
```
This line should have been included in the previous commit.

nixpanic replied:
Ah! yes.

black-dragon74 commented:
If we do not have plans to use `pkg` for anything else, may I suggest a top level `utils` instead?

nixpanic commented:
> If we do not have plans to use `pkg` for anything else, may I suggest a top level `utils` instead?

`pkg` is a common name for go projects, so I decided to take that. In the future, we may be able to move the internal `rbd.Manager` to a public `pkg` package too, so that other tools (rook cli, or other debugging/troubleshooting tools) can use it.

nixpanic commented:
@mergifyio queue

iPraveenParihar commented:
/test ci/centos/mini-e2e/k8s-1.32

nixpanic commented:
/test ci/centos/mini-e2e/k8s-1.32

nixpanic commented:
Might depend on #5193, as testing with the previous Rook version seems to be more stable?

nixpanic commented:
> Might depend on #5193, as testing with the previous Rook version seems to be more stable?

Included the Rook downgrade in this PR too now, just to see if the CI jobs pass.

nixpanic commented:
/retest ci/centos/mini-e2e/k8s-1.32

nixpanic commented:
> /retest ci/centos/mini-e2e/k8s-1.32

Failed with:
```
[FAILED] failed to validate omap count for rados ls --pool=myfs-metadata --namespace=csi | grep -v default | grep -v csi.volume.group. | grep -c ^csi.volume.: expected omap object count 4, got 5
```
[logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.32/detail/mini-e2e_k8s-1.32/109/pipeline)

iPraveenParihar commented:
> > /retest ci/centos/mini-e2e/k8s-1.32
> 
> Failed with:
> 
> ```
> [FAILED] failed to validate omap count for rados ls --pool=myfs-metadata --namespace=csi | grep -v default | grep -v csi.volume.group. | grep -c ^csi.volume.: expected omap object count 4, got 5
> ```
> 
> [logs](https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.32/detail/mini-e2e_k8s-1.32/109/pipeline)


@nixpanic, all jobs have failed at the CephFS test `verifying that ceph-fuse recovery works for new pods`. The NFS test failures are due to stale entries from the failed CephFS test case, so they can be ignored.

I've seen this test case fail previously with mount-utils versions later than v0.29.3 (reference: https://github.com/ceph/ceph-csi/issues/4633). To address this, I had added k8s.io/mount-utils v0.29.3 in the replace section, but it's missing from the main go.mod file, which I suspect is causing the failure. Could you try adding the replace directive in the main go.mod file and see if that resolves the issue?

nixpanic commented:
> @nixpanic, all jobs have failed at the CephFS test `verifying that ceph-fuse recovery works for new pods`. The NFS test failures are due to stale entries from the failed CephFS test case, so they can be ignored.
> 
> I've seen this test case fail previously with mount-utils versions later than v0.29.3 (reference: #4633). To address this, I had added k8s.io/mount-utils v0.29.3 in the replace section, but it's missing from the main go.mod file, which I suspect is causing the failure. Could you try adding the replace directive in the main go.mod file and see if that resolves the issue?

Thanks for the suggestion, @iPraveenParihar! The replacing of `mount-utils` is done in the `e2e/go.mod` file, but not anymore in main `go.mod`. I have added that now again, lets see ho things go :crossed_fingers:

nixpanic commented:
> Thanks for the suggestion, @iPraveenParihar! The replacing of `mount-utils` is done in the `e2e/go.mod` file, but not anymore in main `go.mod`. I have added that now again, lets see ho things go 🤞

Seems to pass with the old mount-utils package!

Rakshith-R commented:
@mergifyio queue
